core/src/test/kotlin/generator/test/GeneratorTest.kt

```diff
@@ -0,0 +1,52 @@
+package generator.test
+
+import generator.TestRunner
+import org.jetbrains.dokka.*
+import org.jetbrains.dokka.utilities.DokkaConsoleLogger
+import org.junit.Test
+import java.nio.file.Path
+
+class GeneratorTest {
+
+    val logger = DokkaConsoleLogger
+    fun config(): (String, Path, Path) -> DokkaConfiguration = { name, root, out ->
+        DokkaConfigurationImpl(
+            outputDir = out.toString(),
+            format = "html",
+            generateIndexPages = true,
+            passesConfigurations = listOf(
+                PassConfigurationImpl(
+                    moduleName = name,
+                    sourceRoots = listOf(root).map { SourceRootImpl(it.toString()) },
+                    analysisPlatform = Platform.jvm,
+                    targets = listOf("jvm")
+                )
+            )
+        )
+    }
```

Comment:
I don't like passing lambdas but I don't have a better way to do it right now, let's discuss it tomorrow

Reply:
Let's leave it like this but please extract those lambda params to some data class and pass that instead

Reply:
Done.